### ISSUE TITLE
ParseObject Pointer data lost when using Parcelable for Activity Transitions

### ISSUE THREAD

**Issue by SkyGanfor** [parse-community/Parse-SDK-Android#952]
I recently ran into an issue when trying to transition between activities and wrap a populated parseobject with parcelable. The parseobject has 40 columns of data mainly consisting of booleans and string, with 3 of them being pointer objects. Before transitioning to the next activity, there is a network query that uses the "include" method to retrieve all the pointers data. But upon transitioning to the next activity using parcelable to wrap the data, all the pointer objects gotten from the "include" method go missing. I can confirm through debugging that the objects are there before wrapping the object in parcelable, and lost immediately after the transition. This issue seems to be tied to the amount of data that exists within the parseobject, as a blank or minimally filled-out object doesn't have this issue. I've included some screenshots of the debug process. Just curious if anyone else has encountered this issue.

![image](https://user-images.githubusercontent.com/22162777/60302745-b3616680-9902-11e9-8218-d1900f23e53e.png)

![image](https://user-images.githubusercontent.com/22162777/60302758-ba887480-9902-11e9-93de-533f48065a85.png)

**Comment by L3K0V** [comment]
I'm into similar situation and you closed the issue @SkyGanfor. What was the problem?

**Comment by SkyGanfor** [comment]
@L3K0V After digging through some of the documentation for Parse Android, I discovered this section that talks about implementing parceling with Parse Object subclasses.
Refer to this: https://docs.parseplatform.org/android/guide/#parceling-parseobject-subclasses

By utilizing the onSaveInstance and onRestoreInstance, any Parse Objects or list of Parse Objects won't be lost when transitioning between activities.

Keep in mind however that if you have local data store enabled in your app, there's currently a bug when trying to use parceling on a list of Parse Objects. The it should cause your app to enter a stackoverflow and lock up. I've created an issue for that bug here: #1003 (hopefully this will be fixed in the future).
